scala-generator/src/main/scala/models/generator/ScalaCaseClasses.scala

```diff
@@ -32,37 +32,45 @@ trait ScalaCaseClasses extends CodeGenerator {
       ""
     }
 
+    val source = s"${header}package ${ssd.namespaces.models} {\n\n  " +
+    Seq(
+      additionalImports.mkString("\n").indent(2),
+      generateCodeBody(ssd).indent(2)
+    ).mkString("\n").trim +
+    s"\n\n}"
+
+    Seq(ServiceFileNames.toFile(ssd.service.namespace, ssd.service.organization.key, ssd.service.application.key, ssd.service.version, "Models", source, Some("Scala")))
+  }
+
+  def generateCodeBody(ssd: ScalaService): String = {
```

Comment:
Created a new function to easily reuse the generator without all the extra bagage.